packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -234,8 +234,11 @@ const MetricForm: FC<MetricFormProps> = ({
   const form = useForm({
     defaultValues: {
       datasource:
-        ("datasource" in current ? current.datasource : datasources[0]?.id) ||
-        "",
+        ("datasource" in current
+          ? current.datasource
+          : settings.defaultDataSource
```

Comment:
For this, the experiment form, and maybe even when we load the general settings form, we only really want to set this if the datasource actually exists right?

How do we handle when the datasource is missing but it's in their org settings? Do we have other org settings that work like this? (for example what happens to the north star metrics when the metric is deleted or the key is updated?) Given this is a reference way may actually want to be a tiny bit careful about this.

Reply:
@lukesonnet Very good call - I just pushed up an update that follows the current convention - if an org attempts to delete a data source that is used for metrics, experiments, etc, we throw an error and tell them they need to remove the connected items before they can delete the data source. So now, an org will get a warning that they need to remove the data source as their org's default before they can delete.

In this update, I also added the ability for a user to clear their default data source, rather than requiring them to select a different one.

Reply:
Thanks! Should we just clear the default in that case instead of erroring?

Reply:
I opted for the error rather than automatically removing it since it follows the current pattern of not deleting/resetting adjacent assets/properties automatically (and silently).

The thought process was the same as when we were thinking about adding the "(Default)" tag next to the data source's name in all of the various modals - to give the user hints that, "hey, this is the data source that someone deemed to be the most common/important/etc.". I think giving a warning that hey, you're about to delete the default data source is a good pattern, though, I see the argument that making a user do that manual work is a bit cumbersome and not the best experience.

In a perfect world, I'd like to add a double confirmation to destructive actions like this

```(e.g. hey, you're about to delete a data source connected to these metrics, experiments, and it's also your organization's default. If you continue, we'll reset your orgs default data source and archive the connected metrics/experiments).```

But, that's a separate project in and of itself.

All of that to say, I trust your judgement as you know our users better than I do, so if you feel strongly that the not-so-good experience of the error is bad enough to warrant just resetting the org's default datasource automatically, its easy to do so.

Reply:
Happy to be consistent and sounds good to me.